Default webcard - Static and Auth Cards

lukedutta9 opened the pull request:
## Summary
<!-- What existing problem does the pull request solve? -->
Adds in ability to create webcards more easily on the app without the need for an app update. To add a new webCard to the app, add a new entry into the default cards json with the following attributes:

```
"cardActive": bool,
"initialURL": String,
"isWebCard": bool,
"requireAuth": bool
```

If the card in the default cards json is not a webCard, then initialURL should have an empty string, and all other attributes should be false.

The following javascript channels are currently supported with the new webcard_container:

1. `OpenLink` - opens links in browser for users
2. `SetHeight` - helps app determine height of webCard (required for the app)
3. `RefreshToken` - refreshes auth token

## Changelog
<!--
    Help reviewers by writing your own changelog entry.

    CATEGORIES: [General, Android, iOS, or Internal]
    TYPES:      [Add, Change, Fix, or Remove]
    MESSAGE:    What and why

    Examples:
    1. [General] [Add] - Add promo banner capability to special events card
    2. [iOS] [Change] - Smooth transitions when navigating between tabs
    3. [Android] [Fix] - Fix a bug causing the user to be logged out
-->
[General][Add] - Add dynamic card creation for webcards

## Test Plan
<!--
    Explain the steps taken to test this update.
    Include screenshots and/or videos if the pull request changes the user interface.
-->
Need to log in to all account types (student, staff, visitor), then go to the home page.
General home page testing like we do for the regression testing will suffice.

kevdesilva01 reviewed (changes requested):
PR functionality looks good. Please deprecate the five < ___>card.dart files that are now redundant for a cleaner PR.

lukedutta9 commented:
Just made a new commit, removed the following cards:

```
lib/ui/campus_info/campus_info_card.dart
lib/ui/campus_info/campus_info_card_backup.dart
lib/ui/student_info/student_info_card.dart
lib/ui/staff_info/staff_info_card.dart
lib/ui/staff_id/staff_id_card.dart
```

kevdesilva01 reviewed (approved):
Cards functionality is preserved, all webcards render with their content and the intended functions all work.

g9singh commented:
SMOKE TEST RESULTS

ANDROID:
- Onboarding login works as expected: PASS
- Profile login works as expected: PASS
- Profile logout works as expected: PASS
- Home view logged out displays visitor cards: PASS
- Home view logged in displays authenticated cards: FAIL (When logging into employee account the MyStudentChart and MyUCSDChart both appear on the home screen)
- Native Scanner Code128 submission successful: PASS
- Native Scanner 2D submission successful: PASS
- Web Scanner Code128 submission successful: PASS
- Web Scanner 2D submission successful: PASS
- DM Push Notification received and appears in the Notifications tab: PASS
- TM Push Notification received and appears in the Notifications tab: PASS

IOS:
- Onboarding login works as expected: PASS
- Profile login works as expected: PASS
- Profile logout works as expected: PASS
- Home view logged out displays visitor cards: PASS
- Home view logged in displays authenticated cards: FAIL (When logging into employee account the MyStudentChart and MyUCSDChart both appear on the home screen)
- Native Scanner Code128 submission successful: PASS
- Native Scanner 2D submission successful: PASS
- Web Scanner Code128 submission successful: PASS
- Web Scanner 2D submission successful: PASS
- DM Push Notification received and appears in the Notifications tab: PASS
- TM Push Notification received and appears in the Notifications tab: PASS

Notes: The feature seems to be working properly except for the MyChart cards. Also, this smoke test was only run on an Employee account. Student accounts should be tested as well.

lukedutta9 commented:
MyCharts card issue stemmed from staff_cards-v3.json, fixed the issue by disabling MyStudentChart for staff_cards-v3.

g9singh commented:
SMOKE TEST 2 RESULTS

ANDROID:

- Onboarding login works as expected: PASS
- Profile login works as expected: PASS
- Profile logout works as expected: PASS
- Home view logged out displays visitor cards: PASS
- Home view logged in displays authenticated cards: PASS
- Native Scanner Code128 submission successful: PASS
- Native Scanner 2D submission successful: PASS
- Web Scanner Code128 submission successful: PASS
- Web Scanner 2D submission successful: PASS
- DM Push Notification received and appears in the Notifications tab: PASS
- TM Push Notification received and appears in the Notifications tab: PASS

IOS:

- Onboarding login works as expected: PASS
- Profile login works as expected: PASS
- Profile logout works as expected: PASS
- Home view logged out displays visitor cards: PASS
- Home view logged in displays authenticated cards: PASS
- Native Scanner Code128 submission successful: PASS
- Native Scanner 2D submission successful: PASS
- Web Scanner Code128 submission successful: PASS
- Web Scanner 2D submission successful: PASS
- DM Push Notification received and appears in the Notifications tab: PASS
- TM Push Notification received and appears in the Notifications tab: PASS

g9singh commented on lib/ui/home/home.dart:
```diff
@@ -67,66 +66,68 @@ class _HomeState extends State<Home> {
+      else {  // dynamically insert webCards into the list
+        orderedCards.add(
+            WebViewContainer(
+                titleText: CardTitleConstants.titleMap[card],
```
You are referencing a constant that is housed within the mobile app source code. This means that an update via the app store and google play store is required anytime we add a new card.

lukedutta9 commented:
Updated PR, the issue with the webcards' titles not being dynamic is now fixed. The default card endpoints have been updated with the new titleText element. In the mobile app, native cards still use the card title constants, but dynamic webCards will be using their own generated card models.

g9singh reviewed (approved):
Passed all smoke tests, and meets requirements outlined in original issue. Well done @lukedutta9.
